code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -366,3 +383,53 @@
 	return
 /mob/living/simple_animal/ExtinguishMob()
 	return
+
+
+//I wanted to call this proc alert but it already exists.
+//Basically makes the mob pay attention to the world, resets sleep timers, awakens it from a sleeping state sometimes
+/mob/living/simple_animal/proc/poke(var/force_wake = 0)
+	if (stat != DEAD)
+		if (force_wake || (!client && prob(30)))
+			wake_up()
+
+//Puts the mob to sleep
+/mob/living/simple_animal/proc/fall_asleep()
+	if (stat != DEAD)
+		resting = 1
+		stat = UNCONSCIOUS
+		canmove = 0
+		wander = 0
+		walk_to(src,0)
+		update_icons()
+
+//Wakes the mob up from sleeping
+/mob/living/simple_animal/proc/wake_up()
+	if (stat != DEAD)
+		stat = CONSCIOUS
+		resting = 0
+		canmove = 1
+		wander = 1
```

Comment:
> Boolean variables and return values should use TRUE and FALSE constans instead of 1 and 0.

https://github.com/discordia-space/CEV-Eris/blob/master/CONTRIBUTING.md